library/core/src/num/nonzero.rs

```diff
@@ -189,3 +190,74 @@ macro_rules! from_str_radix_nzint_impl {
 
 from_str_radix_nzint_impl! { NonZeroU8 NonZeroU16 NonZeroU32 NonZeroU64 NonZeroU128 NonZeroUsize
 NonZeroI8 NonZeroI16 NonZeroI32 NonZeroI64 NonZeroI128 NonZeroIsize }
+
+macro_rules! nonzero_leading_trailing_zeros {
+    ( $( $Ty: ident($Uint: ty) , $LeadingTestExpr:expr ;)+ ) => {
+        $(
+            impl $Ty {
+                doc_comment! {
+                    concat!("Returns the number of leading zeros in the binary representation of `self`.
+On many architectures, this function can perform better than `leading_zeros()` on the underlying integer type, as special handling of zero can be avoided.
```

Comment:
Many tools use the first line/paragraph of the doc comment as a summary. So without an empty line here, the `On many architectures, ..` sentence would end up in the short summary.

(Same for the other three functions.)

```suggestion
                    concat!("Returns the number of leading zeros in the binary representation of `self`.
                    
On many architectures, this function can perform better than `leading_zeros()`
on the underlying integer type, as special handling of zero can be avoided.
```